component/service/src/main/java/org/exoplatform/social/rest/api/ProfileLabelRest.java

```diff
@@ -0,0 +1,274 @@
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import javax.ws.rs.core.Response.Status;
+
+import org.exoplatform.common.http.HTTPStatus;
+import org.exoplatform.commons.ObjectAlreadyExistsException;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.social.core.model.ProfileLabel;
+import org.exoplatform.social.core.profilelabel.ProfileLabelService;
+import org.exoplatform.social.service.rest.api.VersionResources;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+@Path(VersionResources.VERSION_ONE + "/social/profile/label")
+@Tag(name = VersionResources.VERSION_ONE + "/social/profile/label", description = "Operations on labels")
+public class ProfileLabelRest implements ResourceContainer {
+
+  private static final Log          LOG = ExoLogger.getLogger(ProfileLabelRest.class);
+
+  private final ProfileLabelService profileLabelService;
+
+  public ProfileLabelRest(ProfileLabelService profileLabelService) {
+    this.profileLabelService = profileLabelService;
+  }
+
+  @GET
+  @Path("{type}/{id}/{language}")
+  @RolesAllowed("users")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Gets  label for given object and language", method = "GET", description = "This returns label for given object and language")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error") })
+  public Response getLabel(@Context
+  UriInfo uriInfo, @Context
+  Request request,
+                           @Parameter(description = "Object type", required = true)
+                           @PathParam("type")
+                           String type,
+                           @Parameter(description = "Object id", required = true)
+                           @PathParam("id")
+                           String id,
+                           @Parameter(description = "language", required = true)
+                           @PathParam("id")
+                           String language) {
+    try {
+      ProfileLabel profileLabel = profileLabelService.findLabelByObjectTypeAndObjectIdAndLang(type, id, language);
+      return Response.ok(profileLabel).build();
+    } catch (Exception e) {
+      LOG.error("An error occurred while getting list of labels for object type {} and object id {} and language {}",
+                type,
+                id,
+                language,
+                e);
+      return Response.status(HTTPStatus.INTERNAL_ERROR).build();
+    }
+  }
+
+  @GET
+  @Path("labels/{type}/{id}")
```

Comment:
`/social/profile/label/labels/...`

The path has redundancy, please follow REST endpoint [best practices](https://community.exoplatform.com/portal/g/:spaces:engineering/exo_architects/notes/534)